testproject/django_file_form_example/static/example_form_custom_widget.js

```diff
@@ -0,0 +1,55 @@
+const eventEmitter = new EventEmitter3();
+
+function getMetaDataField(fieldName) {
+  // no consideration of prefix??
+  // not sure how to use functions in .utils
+  const name = fieldName + '-metadata';
+  const form = document.getElementById("example-form");
+  return field = form.querySelector(`[name="${name}"]`);
+}
+
+function descriptionChanged(evt) {
+  let field = getMetaDataField(evt.target.getAttribute('fieldName'));
+  let descriptions = document.getElementsByClassName('dff-description');
+  let data = {}
+  for (let i = 0; i < descriptions.length; ++i) {
+    let desc = descriptions[i];
+    data[desc.getAttribute('fileName')] = {
+      'description': desc.value
+    }
+  }
+  field.value = JSON.stringify(data);
+}
+
+eventEmitter.on('addUpload', ({ element, fieldName, upload }) => {
+  // console.log('addUpload', element, fieldName, upload);
+  let field = getMetaDataField(fieldName);
```

Comment:
The metadata field is aliased  so perhaps it is helpful to pass metadata field to user?

Reply:
I now added a `metaDataField` parameter in #344. I also refactored the custom widget code a little.